chia/consensus/blockchain.py

```diff
@@ -802,11 +805,25 @@ async def pre_validate_blocks_multiprocessing(
         self,
         blocks: List[FullBlock],
         npc_results: Dict[uint32, NPCResult],  # A cache of the result of running CLVM, optional (you can use {})
+        sub_slot_iters: Optional[uint64] = None,
+        difficulty: Optional[uint64] = None,
+        prev_ses_block: Optional[BlockRecord] = None,
         batch_size: int = 4,
         wp_summaries: Optional[List[SubEpochSummary]] = None,
         *,
         validate_signatures: bool,
     ) -> List[PreValidationResult]:
+        if difficulty is None or sub_slot_iters is None:
+            block = blocks[0]
+            prev_b = await self.get_block_record_from_db(block.prev_header_hash)
+            sub_slot_iters, difficulty = get_next_sub_slot_iters_and_difficulty(
+                self.constants, len(block.finished_sub_slots) > 0, prev_b, self
+            )
+        if prev_ses_block is None and blocks[0].height > 0:
+            curr = self.block_record(blocks[0].prev_header_hash)
+            while curr.height > 0 and curr.sub_epoch_summary_included is None:
+                curr = self.block_record(curr.prev_hash)
```

Comment:
```suggestion
            curr = self.try_block_record(curr.prev_hash)
            if curr is None:
                return [PreValidationResult(uint16(Err.INVALID_PREV_BLOCK_HASH.value), None, None, False, uint32(0))]
```

Reply:
can this not happen?

Reply:
only as a result of a bug, i am not sure i like the index error but returning INVALID_PREV_BLOCK_HASH would be wrong here